integration-tests/src/test/java/org/apache/druid/tests/indexer/AbstractStreamIndexingTest.java

```
@@ -0,0 +1,439 @@
+      verifyIngestedData(generatedTestConfig);
+    }
+    finally {
+      doMethodTeardown(generatedTestConfig, streamEventWriter);
+    }
+  }
+
+  private void verifyIngestedData(GeneratedTestConfig generatedTestConfig) throws Exception
+  {
+    // Wait for supervisor to consume events
+    LOG.info("Waiting for [%s] millis for stream indexing tasks to consume events", WAIT_TIME_MILLIS);
+    Thread.sleep(WAIT_TIME_MILLIS);
+    // Query data
+    final String querySpec = generatedTestConfig.getStreamQueryPropsTransform().apply(getResourceAsString(QUERIES_FILE));
+    // this query will probably be answered from the indexing tasks but possibly from 2 historical segments / 2 indexing
+    this.queryHelper.testQueriesFromString(querySpec, 2);
+    LOG.info("Shutting down supervisor");
+    indexer.shutdownSupervisor(generatedTestConfig.getSupervisorId());
+    // wait for all indexing tasks to finish
+    LOG.info("Waiting for all indexing tasks to finish");
+    ITRetryUtil.retryUntilTrue(
+        () -> (indexer.getUncompletedTasksForDataSource(generatedTestConfig.getFullDatasourceName()).size() == 0),
+        "Waiting for Tasks Completion"
+    );
+    // wait for segments to be handed off
+    ITRetryUtil.retryUntil(
+        () -> coordinator.areSegmentsLoaded(generatedTestConfig.getFullDatasourceName()),
+        true,
+        10000,
+        30,
+        "Real-time generated segments loaded"
+    );
+
+    // this query will be answered by at least 1 historical segment, most likely 2, and possibly up to all 4
+    this.queryHelper.testQueriesFromString(querySpec, 2);
+  }
+
+  long getSumOfEventSequence(int numEvents)
+  {
+    return (numEvents * (1 + numEvents)) / 2;
+  }
+
+  private void doMethodTeardown(GeneratedTestConfig generatedTestConfig, StreamEventWriter streamEventWriter)
+  {
+    try {
+      streamEventWriter.flush();
+    }
+    catch (Exception e) {
+      // Best effort cleanup as the writer may have already went Bye-Bye
```

"may have already gone"?

---

BTW, I don't think ignoring exceptions is a good idea in any case. Probably it should log at least.

---

hahaha
Done